### PR TITLE
fix(hooks): vm may be null

### PR DESCRIPTION
**Pull request by liulinboyi**
Please make sure these boxes are checked before submitting your PR, thank you!

- [*] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [*] Make sure you are merging your commits to `dev` branch.
- [*] Add some descriptions and refer to relative issues for your PR.

The `getCurrentInstance()`'s result may be null.

### CODE REVIEW

**Review by chenxch — APPROVED** [review]
LGTM🚀

File: packages/hooks/use-prop/index.ts [hunk 1]
```diff
@@ -3,5 +3,5 @@ import type { ComputedRef } from 'vue'
 
 export const useProp = <T>(name: string): ComputedRef<T | undefined> => {
   const vm = getCurrentInstance()!
```
Comment (btea) [inline packages/hooks/use-prop/index.ts:5]: Non-null Assertion can be removed.

File: packages/hooks/use-prop/index.ts [hunk 2]
```diff
@@ -3,5 +3,5 @@ import type { ComputedRef } from 'vue'
...
-  return computed(() => (vm.proxy?.$props as any)[name] ?? undefined)
+  return computed(() => (vm?.proxy?.$props as any)[name] ?? undefined)
```
Comment (plainheart) [inline packages/hooks/use-prop/index.ts:6]: The `?? undefined` seems unnecessary as well.

Comment (liulinboyi) [reply packages/hooks/use-prop/index.ts]: ok, it's removed.

File: packages/hooks/use-prop/index.ts (hunk 1 above)
Comment (liulinboyi) [reply packages/hooks/use-prop/index.ts]: ok, it's removed.

File: packages/hooks/use-prop/index.ts [hunk 3]
```diff
@@ -2,6 +2,6 @@ import { computed, getCurrentInstance } from 'vue'
 import type { ComputedRef } from 'vue'
 
 export const useProp = <T>(name: string): ComputedRef<T | undefined> => {
-  const vm = getCurrentInstance()!
-  return computed(() => (vm.proxy?.$props as any)[name] ?? undefined)
+  const vm = getCurrentInstance()
+  return computed(() => (vm?.proxy?.$props as any)[name])
```
Comment (btea) [inline packages/hooks/use-prop/index.ts:6]: If vm is null, `(vm?.proxy?.$props as any)[name]` may throw an error.

Comment (liulinboyi) [reply packages/hooks/use-prop/index.ts]: ok, chenge to `(vm?.proxy?.$props as any)?.[name]` will resolve the issue.

Test: 
```js
for (let n of [null, { proxy: null }, { proxy: { $props: null } }]) {
    vm = n
    name = 'temp'
    temp = vm?.proxy?.$props?.[name]
    console.log(temp)
}


```

**Review by plainheart — APPROVED** [review]

**Review by btea — APPROVED** [review]

**Pull request merged**